code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+		if(!QDELETED(rift))
+			QDEL_NULL(rift)
+
+/**
+ * Handles wing gust from the windup all the way to the endlag at the end.
+ *
+ * Handles the wing gust attack from start to finish, based on the timer.
+ * When intially triggered, starts at 0.  Until the timer reaches 10, increase Space Dragon's y position by 2 and call back to the function in 1.5 deciseconds.
+ * When the timer is at 10, trigger the attack.  Change Space Dragon's sprite. reset his y position, and push all living creatures back in a 3 tile radius and stun them for 5 seconds.
+ * Stay in the ending state for how much our tiredness dictates and add to our tiredness.
+ * Arguments:
+ * * timer - The timer used for the windup.
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/useGust(timer)
+	if(timer != 10)
+		pixel_y = pixel_y + 2;
+		addtimer(CALLBACK(src, .proc/useGust, timer + 1), 1.5)
+		return
+	pixel_y = 0
+	icon_state = "spacedragon_gust_2"
+	cut_overlays()
+	var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_gust_2")
+	overlay.appearance_flags = RESET_COLOR
+	add_overlay(overlay)
+	playsound(src, 'sound/effects/gravhit.ogg', 100, TRUE)
+	var/gust_locs = spiral_range_turfs(gust_distance, get_turf(src))
+	var/list/hit_things = list()
+	for(var/turf/T in gust_locs)
+		for(var/mob/living/L in T.contents)
+			if(L == src)
+				continue
+			hit_things += L
+			visible_message("<span class='boldwarning'>[L] is knocked back by the gust!</span>")
+			to_chat(L, "<span class='userdanger'>You're knocked back by the gust!</span>")
+			var/dir_to_target = get_dir(get_turf(src), get_turf(L))
+			var/throwtarget = get_edge_target_turf(target, dir_to_target)
+			L.safe_throw_at(throwtarget, 10, 1, src)
+			L.Paralyze(50)
+	addtimer(CALLBACK(src, .proc/reset_status), 4 + ((tiredness * tiredness_mult) / 10))
+	tiredness = tiredness + (gust_tiredness * tiredness_mult)
+
+/**
+  * Sets up Space Dragon's victory for completing the objectives.
+  *
+  * Triggers when Space Dragon completes his objective.
+  * Calls the shuttle with a coefficient of 3, making it impossible to recall.
+  * Sets all of his rifts to allow for infinite sentient carp spawns
+  * Also plays appropiate sounds and CENTCOM messages.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/victory()
+	objective_complete = TRUE
+	permanant_empower()
+	var/datum/antagonist/space_dragon/S = mind.has_antag_datum(/datum/antagonist/space_dragon)
+	if(S)
+		var/datum/objective/summon_carp/main_objective = locate() in S.objectives
+		if(main_objective)
+			main_objective.completed = TRUE
+	priority_announce("A large amount of lifeforms have been detected approaching [station_name()] at extreme speeds. Remaining crew are advised to evacuate as soon as possible.", "Central Command Wildlife Observations")
+	sound_to_playing_players('sound/creatures/space_dragon_roar.ogg')
+	for(var/obj/structure/carp_rift/rift in rift_list)
+		rift.carp_stored = 999999
+		rift.time_charged = rift.max_charge
```

Comment:
Space dragon shouldn't be a round end antag with an immortal carp spawner. There should still be a way to recover

Reply:
It doesn't shuttle call anymore. The carps are also ghost spawn only so it's not effectively infinite.

Reply:
The station can choose to deal with the carp by containing the rift or killing the carp that come out.